docs/standards/generic-standards/lsp1-universal-receiver-delegate.md

```diff
@@ -27,23 +27,34 @@ Therefore, it is advised not to hardcode how the smart contract should handle an
 
 ### Specification
 
-This standard represents a smart contract delegated to the initial `universalReceiver(...)` function.
+:::success recommendation
 
-It contains a single function named `universalReceiverDelegate(...)` that takes the same parameters as the `universalReceiver(...)` function with two additional params:
+Smart contracts implementing the `universalReceiverDelegate(...)` function SHOULD **register** the **[LSP1UniversalReceiverDelegate InterfaceId](../smart-contracts/interface-ids.md) using ERC165**. This way, other contracts can be aware that the contract supports the LSP1 standard.
+
+:::
+
+This standard defines a contract called **UniversalReceiverDelegate** containing a single function named `universalReceiverDelegate(...)` that should be called by the `universalReceiver(..)` function with the `typeId` and `data` passed to it, as well as two other parameters:
 
 - address `caller`: the address that initially called the `universalReceiver(...)` function.
 
 - uint256 `value`: the amount of value sent to the `universalReceiver(...)` function.
 
 ### How Delegation works
 
-The address of the **[UniversalReceiverDelegate](../smart-contracts/lsp1-universal-receiver-delegate-up.md)** contract should be set as a value for the data key [`LSP1UniversalReceiverDelegate`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md#lsp1universalreceiverdelegate), inside the [ERC725Y Data key-value store](https://github.com/ERC725Alliance/erc725/blob/main/docs/ERC-725.md#erc725y) of the implementation contract (contract implementing the `universalReceiver(...)` function).
-
-With such an implementation, the `universalReceiver(...)` function can forward the call to its similar one called `universalReceiverDelegate(...)`.
+The address of the **[UniversalReceiverDelegate](../smart-contracts/lsp1-universal-receiver-delegate-up.md)** contract can be a variable inside the storage, to allow future upgrade of the **UniversalReceiverDelegate** simply by changing the address to another UniversalReceiverDelegate containing a new logic.
```

Comment:
_Suggested rewording:_ (not sure if too wordy, but my ideas was to keep it generic by saying _"inside the contract storage"_ instead of a _"variable"_).

```suggestion
The address of the **[UniversalReceiverDelegate](../smart-contracts/lsp1-universal-receiver-delegate-up.md)** contract can be stored inside the storage of the contract delegating its `universalReceiver(...)` functionality. This allows the upgrade of the **UniversalReceiverDelegate** simply by changing the address (in storage) to another UniversalReceiverDelegate containing a new logic.
```